mule-extensions-api/src/main/java/org/mule/runtime/extension/api/introspection/declaration/type/annotation/DisplayTypeAnnotation.java

```diff
@@ -45,27 +45,31 @@ public String getName() {
   }
 
   /**
-   *
    * @return the name which should be use to render this model.
    */
   public String getDisplayName() {
     return displayModel.getDisplayName();
   }
 
   /**
-   *
    * @return a brief overview about this model.
    */
   public String getSummary() {
     return displayModel.getSummary();
   }
 
+  /**
+   * @return the underlying {@link DisplayModel}
+   */
+  public DisplayModel getDisplayModel() {
+    return displayModel;
+  }
+
   @Override
   public boolean equals(Object obj) {
     if (obj instanceof DisplayTypeAnnotation) {
-      DisplayTypeAnnotation displayTypeAnnotation = ((DisplayTypeAnnotation) obj);
-      return Objects.equals(displayTypeAnnotation.getDisplayName(), this.getDisplayName()) &&
-          Objects.equals(displayTypeAnnotation.getSummary(), this.getSummary());
+      DisplayTypeAnnotation other = ((DisplayTypeAnnotation) obj);
+      return Objects.equals(other.getDisplayModel(), this.getDisplayModel());
```

Comment:
if this is the reason you added the getter above, then it's not needed. You can access the private field directly